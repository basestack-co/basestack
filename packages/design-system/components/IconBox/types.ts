import { SpaceProps } from "styled-system";

export type Size = "small" | "medium" | "large";

export type IconBoxColor = "blue" | "purple" | "gray" | "green";

export type IconBoxVariant = "outlined" | "filled";

export type Gradient = Array<string>;

export interface IconBoxProps extends SpaceProps {
  /**
   * Changes the background color if the variant is outlined
   * */
  backgroundColor?: string;
  /**
   * Changes the icon
   * */
  icon: string;
  /****
   * Icon color
   * */
  color?: IconBoxColor;
  /****
   * Variant filled or outlined
   * */
  variant?: IconBoxVariant;
  /****
   * Changes the border gradient
   * */
  gradient?: Gradient;
  /****
   * Changes the icon color
   * */
  iconColor?: string;
  /****
   * Changes the size
   * */
  size?: Size;
}
