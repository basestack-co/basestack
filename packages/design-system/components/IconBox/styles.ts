import styled from "styled-components";
import { rem } from "polished";

export const IconContainer = styled.div<{ bg: string }>`
  background-color: ${({ bg }) => bg};
  height: ${rem("48px")};
  width: ${rem("48px")};
  border-radius: ${rem("8px")};
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: ${({ theme }) => theme.spacing.s5};
`;
