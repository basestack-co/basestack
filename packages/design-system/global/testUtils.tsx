import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
