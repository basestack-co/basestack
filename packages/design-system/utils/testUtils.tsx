import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
