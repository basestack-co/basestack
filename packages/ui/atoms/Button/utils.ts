import { Size, Target, Type, Variant } from "@/components/atoms/Button/types";
import { css, DefaultTheme } from "styled-components";
import { darken, rem, transparentize } from "polished";

export const handleButtonSize = (size?: Size) => {
  switch (size) {
    case "large":
      return css`
        height: ${rem("60px")};
        font-size: ${rem("18px")};
        line-height: ${rem("25px")};
        padding: 0 ${rem("40px")};
      `;
    case "medium":
      return css`
        height: ${rem("54px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("34px")};
      `;
    case "normal":
      return css`
        height: ${rem("48px")};
        font-size: ${rem("16px")};
        line-height: ${rem("22px")};
        padding: 0 ${rem("30px")};
      `;
    default:
    case "small":
      return css`
        height: ${rem("46px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("24px")};
      `;
    case "xSmall":
      return css`
        height: ${rem("42px")};
        font-size: ${rem("14px")};
        line-height: ${rem("19px")};
        padding: 0 ${rem("20px")};
      `;
  }
};

export const handleButtonsColors = (theme: DefaultTheme) => ({
  primary: {
    outline: {
      color: theme.colors.primary,
      borderColor: theme.colors.primary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.primary,
    },
  },
  secondary: {
    outline: {
      color: theme.colors.secondary,
      borderColor: theme.colors.secondary,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.secondary,
    },
  },
  tertiary: {
    outline: {
      color: theme.colors.gray3,
      borderColor: theme.colors.gray3,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: theme.colors.gray6,
    },
  },
  neutral: {
    outline: {
      color: theme.colors.gray2,
      borderColor: theme.colors.gray2,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.primary,
      borderColor: "transparent",
      backgroundColor: "transparent",
    },
  },
  success: {
    outline: {
      color: theme.colors.success,
      borderColor: theme.colors.success,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.success,
    },
  },
  warning: {
    outline: {
      color: theme.colors.warning,
      borderColor: theme.colors.warning,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.warning,
    },
  },
  danger: {
    outline: {
      color: theme.colors.error,
      borderColor: theme.colors.error,
      backgroundColor: "transparent",
    },
    filled: {
      color: theme.colors.white,
      borderColor: "transparent",
      backgroundColor: theme.colors.error,
    },
  },
});

export const darkenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return darken(
    amount || 0.1,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};

export const lightenColors = (
  theme: DefaultTheme,
  amount: number,
  type: Type,
  variant: Variant,
  target: Target
) => {
  return transparentize(
    amount || 0.6,
    handleButtonsColors(theme)[type || "primary"][variant || "filled"][target]
  );
};
