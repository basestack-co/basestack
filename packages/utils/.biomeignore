# Node dependencies
node_modules

# Next.js build output
.next
out

# Environment files
.env
.env.*
!.env.example

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*

# OS / editor files
.DS_Store
Thumbs.db

# Package manager cache
.pnp
.pnp.js
.pnp.cjs

# Test coverage
coverage

# Build artifacts
dist
build

# Temporary files
.tmp
temp