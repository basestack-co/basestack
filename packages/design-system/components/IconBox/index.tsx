import { transparentize } from "polished";
import React from "react";
import { useTheme } from "styled-components";
import Icon from "../Icon";
import { ContentContainer, IconContainer } from "./styles";
import { IconBoxColor, IconBoxProps, IconBoxVariant } from "./types";

const IconBox = ({
  icon,
  color = "gray",
  variant = "outlined",
  backgroundColor,
  iconColor,
  gradient,
  size = "large",
  ...props
}: IconBoxProps) => {
  const { colors, isDarkMode } = useTheme();

  const lightStyles = {
    green: {
      iconColor: colors.green500,
      backgroundColor: colors.green50,
    },
    blue: {
      iconColor: colors.blue500,
      backgroundColor: colors.blue50,
    },
    purple: {
      iconColor: colors.purple500,
      backgroundColor: colors.purple50,
    },
    gray: {
      iconColor: colors.gray500,
      backgroundColor: colors.gray50,
    },
  };

  const darkStyles = {
    green: {
      iconColor: colors.green200,
      backgroundColor: transparentize(0.8, colors.green200),
    },
    blue: {
      iconColor: colors.blue200,
      backgroundColor: transparentize(0.8, colors.blue200),
    },
    purple: {
      iconColor: colors.purple200,
      backgroundColor: transparentize(0.8, colors.purple200),
    },
    gray: {
      iconColor: colors.gray300,
      backgroundColor: transparentize(0.8, colors.gray200),
    },
  };

  const styles = isDarkMode ? darkStyles : lightStyles;

  const iconColorVariant =
    styles[variant === "outlined" ? "gray" : color].iconColor;

  return (
    <IconContainer
      className="icon-box"
      variant={variant}
      filledBg={styles[color].backgroundColor}
      outlinedBg={backgroundColor}
      gradient={gradient}
      size={size}
      {...props}
    >
      <ContentContainer>
        <Icon
          icon={icon}
          color={iconColor || iconColorVariant}
          size={size === "small" ? "small" : "medium"}
        />
      </ContentContainer>
    </IconContainer>
  );
};

export type { IconBoxProps, IconBoxColor, IconBoxVariant };
export default IconBox;
