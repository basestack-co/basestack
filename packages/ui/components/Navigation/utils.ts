// Types
import { Translate } from "next-translate";
import { NextRouter } from "next/router";
// Utils
import { config } from "@basestack/utils";
import { signOut } from "next-auth/react";

export const getInternalLinks = (t: Translate, formId: string) => {
  return [
    {
      icon: "flag",
      activeText: "Submissions",
      to: `/form/${formId}/submissions`,
      text: t("internal.submissions"),
    },
    {
      icon: "flag",
      activeText: "Setup",
      to: `/form/${formId}/setup`,
      text: t("internal.setup"),
    },
    {
      icon: "settings",
      activeText: "Settings",
      to: `/form/${formId}/settings/general`,
      text: t("internal.settings"),
    },
  ];
};

export const getExternalLinks = (t: Translate) => {
  return [
    {
      icon: "description",
      activeText: "Documentation",
      to: config.urls.docs.flags.base,
      isExternal: true,
      text: t("external.docs"),
    },
    {
      icon: "link",
      activeText: "Github",
      to: config.urls.repo,
      isExternal: true,
      text: t("external.github"),
    },
  ];
};

export const getAvatarDropdownList = (
  t: Translate,
  router: NextRouter,
  onCreateForm: () => void,
) => {
  return [
    {
      icon: "add_circle",
      text: t("create.form"),
      onClick: onCreateForm,
      separator: true,
    },
    {
      icon: "settings",
      text: t("dropdown.settings"),
      onClick: () =>
        router.push({
          pathname: "/user/profile/settings",
        }),
    },
    {
      icon: "logout",
      text: t("dropdown.logout"),
      onClick: signOut,
    },
  ];
};
