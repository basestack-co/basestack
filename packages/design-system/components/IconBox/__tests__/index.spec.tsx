import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import IconBox from "..";

describe("IconBox tests", () => {
  afterEach(cleanup);

  test("should render IconBox blue", () => {
    const { asFragment } = renderWithTheme(
      <IconBox icon="help" color="blue" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render IconBox purple", () => {
    const { asFragment } = renderWithTheme(
      <IconBox icon="help" color="purple" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render IconBox gray", () => {
    const { asFragment } = renderWithTheme(
      <IconBox icon="help" color="gray" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render IconBox green", () => {
    const { asFragment } = renderWithTheme(
      <IconBox icon="help" color="green" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
