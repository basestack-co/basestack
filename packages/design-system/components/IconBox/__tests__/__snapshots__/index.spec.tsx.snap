// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBox tests should render IconBox blue 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c3 {
  color: #757575;
  font-size: 1.5rem;
}

.c0 {
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c0::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #FFFFFF;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c1 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

<div
    class="c0 icon-box"
    variant="outlined"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
      >
        <span
          class="c3 material-symbols-rounded"
          data-testid="icon"
        >
          help
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IconBox tests should render IconBox gray 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c3 {
  color: #757575;
  font-size: 1.5rem;
}

.c0 {
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c0::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #FFFFFF;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c1 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

<div
    class="c0 icon-box"
    variant="outlined"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
      >
        <span
          class="c3 material-symbols-rounded"
          data-testid="icon"
        >
          help
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IconBox tests should render IconBox green 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c3 {
  color: #757575;
  font-size: 1.5rem;
}

.c0 {
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c0::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #FFFFFF;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c1 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

<div
    class="c0 icon-box"
    variant="outlined"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
      >
        <span
          class="c3 material-symbols-rounded"
          data-testid="icon"
        >
          help
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IconBox tests should render IconBox outlined 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c3 {
  color: #757575;
  font-size: 1.5rem;
}

.c0 {
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c0::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #FFFFFF;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c1 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

<div
    class="c0 icon-box"
    variant="outlined"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
      >
        <span
          class="c3 material-symbols-rounded"
          data-testid="icon"
        >
          help
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IconBox tests should render IconBox purple 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c3 {
  color: #757575;
  font-size: 1.5rem;
}

.c0 {
  position: relative;
  height: 3rem;
  width: 3rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(220deg,          #F6F6F6 0%,          #CBCBCB 30%,          #AFAFAF 70%,          #F6F6F6 100%);
}

.c0::before {
  content: "";
  position: absolute;
  inset: 1px;
  background: #FFFFFF;
  border-radius: 0.4375rem;
  z-index: 0;
}

.c1 {
  position: relative;
  display: inline-flex;
  z-index: 1;
}

<div
    class="c0 icon-box"
    variant="outlined"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="icon-container"
      >
        <span
          class="c3 material-symbols-rounded"
          data-testid="icon"
        >
          help
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
